### PR TITLE
sql-statements: use EBNF to render syntax diagrams - second batch (#5376)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #5376 to release-2.1
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In docs-cn repo:
git pr https://github.com/pingcap/docs-cn/pull/5406
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/docs-cn.git pr/5406:release-2.1-541d0a6b1ba7
```

---

### What is changed, added or deleted? (Required)

This PR changes the synopsis sections of BEGIN, CHANGE, COMMIT, CREATE, DEALLOCATE, DELETE, DO, and DROP statements to use the ` ```ebnf+diagrm ` block, replacing the original image-based sqlgrams. It is a follow-up PR to https://github.com/pingcap/docs/issues/4389. 

To avoid making one single PR too large, I will open 4-5 follow-up PRs to process sql-statements files in batches. And I will delete all image files in the last PR in case of any display issues.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://pingcap.github.io/sqlgram

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Need modification after applied to another branch 
- [x] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sql-statements/sql-statement-create-database.md [hunk 1]
```diff
@@ -10,25 +10,22 @@ aliases: ['/docs-cn/v2.1/sql-statements/sql-statement-create-database/','/docs-c
...
 ## 语法图
 
-**CreateDatabaseStmt:**
+```ebnf+diagram
+CreateDatabaseStmt ::=
+    'CREATE' 'DATABASE' IfNotExists DBName DatabaseOptionListOpt
 
-![CreateDatabaseStmt](/media/sqlgram/CreateDatabaseStmt.png)
+DatabaseSym ::=
+    DATABASE
```
Comment (CharLotteiu) [inline sql-statements/sql-statement-create-database.md:18]: These lines are newly added. Please take a look~ @kennytm

File: sql-statements/sql-statement-create-index.md [hunk 2]
```diff
@@ -10,37 +10,38 @@ aliases: ['/docs-cn/v2.1/sql-statements/sql-statement-create-index/','/docs-cn/v
...
 ## 语法图
 
-**CreateIndexStmt:**
+```ebnf+diagram
+CreateIndexStmt ::=
+    'CREATE' IndexKeyTypeOpt 'INDEX' IfNotExists Identifier IndexTypeOpt 'ON' TableName '(' IndexPartSpecificationList ')' IndexOptionList IndexLockAndAlgorithmOpt
 
-![CreateIndexStmt](/media/sqlgram/CreateIndexStmt.png)
+CreateIndexStmtUnique ::=
+    UNIQUE?
```
Comment (CharLotteiu) [inline sql-statements/sql-statement-create-index.md:18]: Same as the previous comment.

File: sql-statements/sql-statement-create-index.md [hunk 3]
```diff
@@ -10,37 +10,38 @@ aliases: ['/docs-cn/v2.1/sql-statements/sql-statement-create-index/','/docs-cn/v
...
+IndexTypeOpt ::=
+    IndexType?
 
-**Identifier:**
+TableName ::=
+    Identifier ('.' Identifier)?
 
-![Identifier](/media/sqlgram/Identifier.png)
+IndexColNameList ::=
+    IndexColName (',' IndexColName)*
```
Comment (CharLotteiu) [inline sql-statements/sql-statement-create-index.md:33]: Same as the previous comment.

File: sql-statements/sql-statement-create-table.md [hunk 4]
```diff
@@ -10,49 +10,43 @@ aliases: ['/docs-cn/v2.1/sql-statements/sql-statement-create-table/','/docs-cn/v
...
-
-**ColumnOptionListOpt:**
-
-![ColumnOptionListOpt](/media/sqlgram/ColumnOptionListOpt.png)
-
-**TableOptionListOpt:**
-
-![TableOptionListOpt](/media/sqlgram/TableOptionListOpt.png)
+TableOptionListOpt ::=
+    '' | TableOption* ( ',' TableOption* )*
```
Comment (CharLotteiu) [inline sql-statements/sql-statement-create-table.md:48]: @kennytm Please take a look at these lines as well. Thanks!

File: sql-statements/sql-statement-create-database.md (hunk 1 above)
Comment (kennytm) [reply sql-statements/sql-statement-create-database.md:18]: it should be `DatabaseSym ::= 'DATABASE' | 'SCHEMA'`, e.g. `CREATE SCHEMA ...` is a valid CreateDatabaseStmt.

File: sql-statements/sql-statement-create-index.md (hunk 2 above)
Comment (kennytm) [reply sql-statements/sql-statement-create-index.md:18]: i guess this should be inlined into `IndexType`. BTW `'UNIQUE'` not `UNIQUE`.

File: sql-statements/sql-statement-create-index.md (hunk 3 above)
Comment (kennytm) [reply sql-statements/sql-statement-create-index.md:33]: superseded by `IndexPartSpecificationList`.

Comment (CharLotteiu) [reply sql-statements/sql-statement-create-index.md:33]: Is it alright to use `IndexPartSpecificationList` in docs for release 2.1?

Comment (kennytm) [reply sql-statements/sql-statement-create-index.md:33]: if `CREATE INDEX i ON t (col(123));` works on 2.1 then it should be IndexPartSpecificationList.

if not the `IndexPartSpecificationList` in `CreateIndexStmt` should be replaced by `IndexColNameList`.

File: sql-statements/sql-statement-create-table.md (hunk 4 above)
Comment (kennytm) [reply sql-statements/sql-statement-create-table.md:48]: lolwut 🤣 

```suggestion
TableOptionListOpt ::=
    (TableOption (','? TableOption)*)?
```

also `CreateTableOptionListOpt` is equivalent to `TableOptionListOpt`.

**Comment by CharLotteiu** [comment]
Closed as we decided not to maintain the EBNF code of synopsis on release-2.1 branch.

**Pull request closed without merging**